api.go

```diff
@@ -2892,9 +2892,40 @@ func (sc *Client) GetSitePrimaryAgentKey(corpName, siteName string) (PrimaryAgen
 	return primaryKey, nil
 }
 
+func (sc *Client) doRequestDetailed(method, url, reqBody string) (*http.Response, error) {
```

Comment:
This appears to duplicate much of `doRequest`. Can be the first part of that function just be calling `doRequestDetailed` so there's no concern about keeping them in sync?

Reply:
`doRequest` and `doRequestDetailed` return different data types. I would love to use just `doRequest`, but I do not have sufficient knowledge about the intended design of this SDK to have an informed opinion on to do that level of refactoring without the risk of breaking other functions.

Do you have a suggested approach?

Reply:
@BrooksCunningham I think what @mattg-sigsci is suggesting is to replace [this block](https://github.com/signalsciences/go-sigsci/blob/main/api.go#L78-L105) in `doRequest` with a call to `doRequestDetailed`, something like this:

```
func (sc *Client) doRequest(method, url, reqBody string) ([]byte, error) {
    resp, err := doRequestDetailed(method, url, reqBody)
    if err != nil {
        return []byte{}, err
    }
    defer resp.Body.Close()

    body, err := io.ReadAll(resp.Body)
    if err != nil {
        return []byte{}, err
    }

    switch method {
    case "GET":
        if resp.StatusCode != http.StatusOK {
            return body, errMsg(body)
        }
    case "POST":
        switch resp.StatusCode {
        case http.StatusOK, http.StatusCreated, http.StatusNoContent:
        default:
            return body, errMsg(body)
        }
    case "DELETE":
        switch resp.StatusCode {
        case http.StatusOK, http.StatusNoContent:
        default:
            return body, errMsg(body)
        }
    case "PUT":
        fallthrough
    case "PATCH":
        if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
            return body, errMsg(body)
        }
    }

    return body, nil
}
```

Reply:
Thank you for the example. The reason that I return the entire response object with `doRequestDetailed` is because I want access to the response status codes and response body. The response status code is a very useful identifier to know when a request should be retried. I could not find a consistent way to know when to retry a request solely on the response body, but I may have missed something.

@shawnps Do you have a proposed way to know when to retry requests to the function `CreateOrUpdateEdgeDeploymentService` using your above example?

Reply:
I think I understand the goal after having a chat with @vvuksan . Thanks @vvuksan !

I'll make the suggested update. Thank you!